USAGE.md

```diff
@@ -1,370 +1,707 @@
-import stix2
+from stix2.utils import get_type_from_id
+
+def get_techniques_by_group_software(src, group_stix_id):
+    # get the malware, tools that the group uses
+    group_uses = [
+        r for r in src.relationships(group_stix_id, 'uses', source_only=True)
+        if get_type_from_id(r.target_ref) in ['malware', 'tool']
+    ]
+
+    # get the technique stix ids that the malware, tools use
+    software_uses = src.query([
+        Filter('type', '=', 'relationship'),
+        Filter('relationship_type', '=', 'uses'),
+        Filter('source_ref', 'in', [r.source_ref for r in group_uses])
+    ])
+
+    #get the techniques themselves
+    return src.query([
+        Filter('type', '=', 'attack-pattern'),
+        Filter('id', 'in', [r.target_ref for r in software_uses])
+    ])
+
+get_techniques_by_group_software(fs, "intrusion-set--f047ee18-7985-4946-8bfb-4ed754d3a0dd")
+```
+
+## Working with deprecated and revoked objects
+Objects that are deemed no longer beneficial to track as part of the knowledge base are marked as deprecated, and objects which are replaced by a different object are revoked. In both cases, the old object is marked with a field (either `x_mitre_deprecated` or `revoked`) noting their status. In the case of revoked objects, a relationship of type `revoked-by` is also created targeting the replacing object. 
+
+### Removing revoked and deprecated objects
+Revoked and deprecated objects are kept in the knowledge base so that workflows relying on those objects are not 
+broken. We recommend you filter out revoked and deprecated objects from your views whenever possible since they are no 
+longer maintained by ATT&CK.
+
+Revoked and deprecated objects can be removed quite easily:
+
+```python
+def remove_revoked_deprecated(stix_objects):
+    """Remove any revoked or deprecated objects from queries made to the data source"""
+    # Note we use .get() because the property may not be present in the JSON data. The default is False
+    # if the property is not set.
+    return list(
+        filter(
+            lambda x: x.get("x_mitre_deprecated", False) is False and x.get("revoked", False) is False,
+            stix_objects
+        )
+    )
+
+mitigations = src.query([ stix2.Filter("type", "=", "course-of-action") ])
+mitigations = remove_revoked_deprecated(mitigations)
+```
 
-enterprise_attack_fs = stix2.FileSystemSource("./cti/enterprise-attack")
-pre_attack_fs = stix2.FileSystemSource("./cti/pre-attack")
-mobile_attack_fs = stix2.FileSystemSource("./cti/mobile-attack")
 
-composite_ds = stix2.CompositeDataSource()
-composite_ds.add_data_sources([enterprise_attack_fs, pre_attack_fs, mobile_attack_fs])
+### Getting a revoking object
+When an object is replaced by another object, it is marked with the field `revoked` a relationship of type `revoked-by` is created to where the `source_ref` is the revoked object and the `target_ref` is the revoking object. This relationship can be followed to find the replacing object:
```

Comment:
Small grammatical mishap? "When an object is replaced by another object, it is marked with the field `revoked` **and** a relationship of type `revoked-by` is created _[removed]_ where the `source_ref` is the revoked object and the `target_ref` is the revoking object."